src/middleware/userAdminAccessMiddleware.js

```diff
@@ -0,0 +1,26 @@
+import httpCodes from 'http-codes';
+import { validateUserAuthForAdmin } from '../services/accessValidation';
+import handleErrors from '../lib/apiErrorHandler';
+
+/**
+ * Admin Access Middleware
+ *
+ * This middleware handles the user admin access by delegating to the
+ * admin access validation service. At this point a user is logged in,
+ * but might not have the admin priviledges.
+ * @param {*} req - request
+ * @param {*} res - response
+ * @param {*} next - next middleware
+ */
+export default async function userAdminAccessMiddleware(req, res, next) {
+  try {
+    if (!(await validateUserAuthForAdmin(req))) {
+      // send a 404 rather than a 403 (Forbidden) to avoid confirming route
```

Comment:
:+1:

Reply:
Can we talk about this decision? I think I'd rather send the `403`.

Reply:
No problem. Let me set 404 for now and we can discuss.